src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -1472,37 +1484,45 @@ namespace
                 AABB2i(
                     Vector2i(x0, y0),
                     Vector2i(x1, y1)));
+            m_attribute_editor->refresh();
         }
 
       private:
         const QRect m_rect;
+        const AttributeEditor* m_attribute_editor;
     };
 }
 
 void MainWindow::slot_clear_render_region()
 {
-    m_rendering_manager.schedule(
-        auto_ptr<RenderingManager::IScheduledAction>(
-            new ClearRenderRegionAction()));
+    auto_ptr<RenderingManager::IScheduledAction> clear_render_region_action(
+        new ClearRenderRegionAction(m_attribute_editor));
 
-    m_rendering_manager.reinitialize_rendering();
+    if (m_rendering_manager.is_rendering())
+        m_rendering_manager.schedule(clear_render_region_action);
+    else clear_render_region_action.get()->operator()(
+        *m_project_manager.get_project());
 
-    emit signal_crop_window_cleared();
+    m_rendering_manager.reinitialize_rendering();
 }
 
 void MainWindow::slot_set_render_region(const QRect& rect)
 {
-    m_rendering_manager.schedule(
-        auto_ptr<RenderingManager::IScheduledAction>(
-            new SetRenderRegionAction(rect)));
+    auto_ptr<RenderingManager::IScheduledAction> set_render_region_action(
+        new SetRenderRegionAction(rect, m_attribute_editor));
 
-    if (m_settings.get_path_optional<bool>(SETTINGS_RENDER_REGION_TRIGGERS_RENDERING))
+    if (m_settings.get_path_optional<bool>(SETTINGS_RENDER_REGION_TRIGGERS_RENDERING)
+        && !m_rendering_manager.is_rendering())
+    {
+        m_rendering_manager.schedule(set_render_region_action);
```

Comment:
Scheduling an action and starting a rendering process straight after was a part of original logic. It sounds legit, since renderer starts the batch job from iteration over scheduled actions. 
But I can change it, no worries.
And you are right, there should be 3 cases:
1. Execute an action if we are not rendering and we are not going to by the settings (the one missed in original code)
2. Schedule an action if we are rendering (the one missed by me)
3. Schedule or force an action and start rendering if corresponding value is set in options